bump.go

```diff
@@ -0,0 +1,100 @@
+package paymail
+
+import (
+	"errors"
+	"github.com/libsv/go-bc"
+)
+
+// BUMPs represents a slice of BUMPs - BSV Unified Merkle Paths
+type BUMPs []BUMP
+
+// BUMP is a struct that represents a whole BUMP format
+type BUMP struct {
+	blockHeight uint64
+	path        [][]BUMPLeaf
+}
+
+// BUMPLeaf represents each BUMP path element
+type BUMPLeaf struct {
+	hash      string
+	txId      bool
+	duplicate bool
+	offset    uint64
+}
+
+func (b BUMP) calculateMerkleRoots() ([]string, error) {
+	merkleRoots := make([]string, 0)
+
+	for _, bumpPathElement := range b.path[0] {
+		merkleRoot, err := calculateMerkleRoot(bumpPathElement, b)
+		if err != nil {
+			return nil, err
+		}
+		merkleRoots = append(merkleRoots, merkleRoot)
+	}
+	return merkleRoots, nil
+}
+
+func findTxByOffset(offset uint64, bumpLeaves []BUMPLeaf) *BUMPLeaf {
+	for _, bumpTx := range bumpLeaves {
+		if bumpTx.offset == offset {
+			return &bumpTx
+		}
+	}
+	return nil
+}
+
+// calculateMerkleRoots will calculate one merkle root for tx in the BUMPPath
+func calculateMerkleRoot(baseTx BUMPLeaf, bump BUMP) (string, error) {
```

Comment:
Please change names from `*Tx` to `*Leaf`

eg. `baseTx => baseLeaf`, `tx2 => leafInPair (?)`

Reply:
Thanks, I forgot about it. Done